hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/AggregatePhysicalRule.java

```diff
@@ -71,9 +73,18 @@ public void onMatch(RelOptRuleCall call) {
     }
 
     private static RelNode optimize(AggregateLogicalRel logicalAggregate, RelNode physicalInput) {
-        return logicalAggregate.getGroupSet().cardinality() == 0
-                ? toAggregate(logicalAggregate, physicalInput)
-                : toAggregateByKey(logicalAggregate, physicalInput);
+        HazelcastRelMetadataQuery query = HazelcastRelMetadataQuery.reuseOrCreate(physicalInput.getCluster().getMetadataQuery());
+        WindowProperties windowProperties = query.extractWindowProperties(physicalInput);
+        if (windowProperties == null) {
+            return logicalAggregate.getGroupSet().cardinality() == 0
+                    ? toAggregate(logicalAggregate, physicalInput)
+                    : toAggregateByKey(logicalAggregate, physicalInput);
+        } else {
+            WindowProperties.WindowProperty windowProperty = windowProperties.findFirst(logicalAggregate.getGroupSet().asList());
+            return windowProperty == null
+                    ? toStreamingAggregate(logicalAggregate, physicalInput)
+                    : toSlidingAggregateByKey(logicalAggregate, physicalInput, windowProperty);
+        }
```

Comment:
This is not correct. If group set cardinality is 0 (we're not grouping), it should fail _iff_ it's a stream. If we're grouping, it should fail if the input is a stream and if we're not grouping by a window property.

The query in `test_noGroupBy()` should pass.